### ISSUE TITLE
ERROR: bamShrink required for SV calling

### ISSUE THREAD

**Issue by malmarri** [DecodeGenetics/graphtyper-pipelines#6]

**Comment by malmarri** [comment]
Hi,

Just read your paper (congrats) and would like to try your tool. I have a few hundred WGS 30x samples, I used manta to discover SV and merged them with svimmer. I'm only interested in SVs and trying to genotype using graphtyper, however I'm having problems running make_graphtyper_sv_pipeline.sh. I first tried the test files, pipelines.make, which I had an error while running, due to an intermediate vcf not being sorted:

..
[D|D|G|G|G] Completed slice 20:500-4999
[D|D|G|G|G] Completed slice 20:5000-9499
[E::hts_idx_push] Unsorted positions on sequence #1: 5237 followed by 4965
tbx_index_build failed: /tmp/graphtyper_calling.iVr84N/final_hap_calls.vcf.gz
make: *** [test] Error 1

I modified the node_script.sh removing the --no-sort option and it seems to work:
$GRAPHTYPER vcf_concatenate $TMP/results/${chrom}/*.vcf.gz {--no_sort} --output=$TMP/final_small_variants.vcf.gz

[D|D|G|G|G] Completed slice 20:500-4999
[D|D|G|G|G] Completed slice 20:5000-9499

Graphtyper finished genotyping the region successfully. Final results are at:
results/20/000000500-000009499.vcf.gz

Total wall-clock time of genotyping with Graphtyper (seconds):
2

Total time (seconds):
3

I then tried the SV calling, I followed the README file in the test_sv folder, and was able to genotype the test files correctly with no error. However, once I try to genotype my samples using the make_graphtyper_sv_pipeline.sh - after setting all the correct paths - there is no process running, it just exits with no error. I checked all the individual .sh files, but can't seem to find the issue. Could you please assist me with this issue.

Best wishes,
Mo

**Comment by malmarri** [comment]
I think I've sorted this, i'll close the issue. Thanks

**Comment by malmarri** [comment]
Hi,

Just read your paper (congrats) and would like to try your tool. I have a few hundred WGS 30x samples, I used manta to discover SV and merged them with svimmer. I'm only interested in SVs and trying to genotype using graphtyper, however I'm having problems running make_graphtyper_sv_pipeline.sh. I first tried the test files, pipelines.make, which I had an error while running, due to an intermediate vcf not being sorted:

..
[D|D|G|G|G] Completed slice 20:500-4999
[D|D|G|G|G] Completed slice 20:5000-9499
[E::hts_idx_push] Unsorted positions on sequence #1: 5237 followed by 4965
tbx_index_build failed: /tmp/graphtyper_calling.iVr84N/final_hap_calls.vcf.gz
make: *** [test] Error 1

I modified the node_script.sh removing the --no-sort option and it seems to work:
$GRAPHTYPER vcf_concatenate $TMP/results/${chrom}/*.vcf.gz {--no_sort} --output=$TMP/final_small_variants.vcf.gz

[D|D|G|G|G] Completed slice 20:500-4999
[D|D|G|G|G] Completed slice 20:5000-9499

Graphtyper finished genotyping the region successfully. Final results are at:
results/20/000000500-000009499.vcf.gz

Total wall-clock time of genotyping with Graphtyper (seconds):
2

Total time (seconds):
3

I then tried the SV calling, I followed the README file in the test_sv folder, and was able to genotype the test files correctly with no error. However, once I try to genotype my samples using the make_graphtyper_sv_pipeline.sh - after setting all the correct paths - the runs exit with "ERROR: bamShrink required for SV calling". I can't seem to find find "bamShrink" in the config files?

Best wishes,
Mo

**Comment by malmarri** [comment]
Hi,

Just read your paper (congrats) and would like to try your tool. I have a few hundred WGS 30x samples, I used manta to discover SV and merged them with svimmer. I'm only interested in SVs and trying to genotype using graphtyper, however I'm having problems running make_graphtyper_sv_pipeline.sh. I first tried the test files, pipelines.make, which I had an error while running, due to an intermediate vcf not being sorted:

..

> 
> [D|D|G|G|G] Completed slice 20:500-4999
> [D|D|G|G|G] Completed slice 20:5000-9499
> [E::hts_idx_push] Unsorted positions on sequence #1: 5237 followed by 4965
> tbx_index_build failed: /tmp/graphtyper_calling.iVr84N/final_hap_calls.vcf.gz
> make: *** [test] Error 1

I modified the node_script.sh removing the --no-sort option and it seems to work:
`$GRAPHTYPER vcf_concatenate $TMP/results/${chrom}/*.vcf.gz {--no_sort} --output=$TMP/final_small_variants.vcf.gz
`

> [D|D|G|G|G] Completed slice 20:500-4999
> [D|D|G|G|G] Completed slice 20:5000-9499
> 
> Graphtyper finished genotyping the region successfully. Final results are at:
> results/20/000000500-000009499.vcf.gz
> 
> Total wall-clock time of genotyping with Graphtyper (seconds):
> 2
> 
> Total time (seconds):
> 3
> 

I then tried the SV calling, I followed the README file in the test_sv folder, and was able to genotype the test files correctly with no error. However, once I try to genotype my samples using the make_graphtyper_sv_pipeline.sh - after setting all the correct paths - the runs exit with "ERROR: bamShrink required for SV calling". I can't seem to find "bamShrink" in the config files?

Best wishes,
Mo
